File: android/catalog/src/main/kotlin/com/cacompte/catalog/generic/GenericSumRules.kt
package com.cacompte.catalog.generic

import com.cacompte.domain.rules.GameRules

/**
 * Miroir de `GenericSumRules.swift` — corps vide, hérite entièrement des méthodes par défaut de
 * [GameRules] (14 des 20 jeux du catalogue, voir `spec/games (fichiers .json)`, champ `"engine"`).
 */
class GenericSumRules : GameRules {
    override val engineID: String = "generic.sum.v1"

    companion object {
        const val ENGINE_ID = "generic.sum.v1"
    }
}

File: apple/QuiMeneKit/Sources/Catalog/GenericRules/GenericSumRules.swift
import Domain

/// Doc 04 : « un jeu simple n'implémente rien du tout. » Les quatre méthodes viennent
/// entièrement de l'extension par défaut de `GameRules`. Sert tous les jeux dont
/// `"engine": "generic.sum.v1"`.
public struct GenericSumRules: GameRules {
  public static let engineID = "generic.sum.v1"

  public init() {}
}
